sirepo/job_supervisor.py

```diff
@@ -809,3 +854,75 @@ def send(self):
 
     def __hash__(self):
         return hash((self.opId,))
+
+
+def _purge_free_simulations_set():
+    def tomorrow_2am_mst():
+        if pkconfig.channel_in_internal_test():
+            return sirepo.srtime.utc_now_as_float() \
+                + cfg.test_purge_frequency_secs
+        # 9am UTC == 2am MST: Does not adjust for daylight savings
+        t = sirepo.srtime.utc_now() + datetime.timedelta(days=1)
+        return sirepo.srtime.to_timestamp(
+          datetime.datetime(t.year, t.month, t.day, 9, 0, 0, 0)
+        )
+
+    tornado.ioloop.IOLoop.current().call_at(
+        tomorrow_2am_mst(),
+        _purge_free_simulations,
+    )
+
+
+async def _purge_free_simulations():
+    async def _get_uids_and_files():
+        r = PKDict()
+        p = sirepo.auth_db.UserRole.uids_of_paid_users()
+        for f in pkio.sorted_glob(_DB_DIR.join('*')):
+            u = str(f.basename).split('-')[0]
+            if u in p or not _is_older_than_days(f):
+                continue
+            if u not in r:
+                r[u] = []
+                await tornado.gen.sleep(0) # TODO(e-carlin): necessary?
+            r[u].append(f)
+        return r
+
+    def _get_run_dir_path(data):
+        data.report = data.computeModel
+        return sirepo.simulation_db.simulation_run_dir(data)
+
+    def _is_older_than_days(db_file):
+        return db_file.mtime() < (
+            sirepo.srtime.utc_now_as_float() -
+            (cfg.purge_free_after_days * 24 * 60 * 60)
+        )
+
+    def _purge_sim(db_file):
+        # OPTIMIZATION: We assume the uids_of_paid_users doesn't change very
+        # frequently so we don't need to check again. A user could run a sim
+        # at anytime so we need to check that they haven't
+        if not _is_older_than_days(db_file):
+            return
+        d = pkcollections.json_load_any(db_file)
+        if d.status == job.FREE_USER_PURGED:
+            # TODO(e-carlin): should we touch the file so we don't look at again for a bit?
```

Comment:
We should touch the file, and indeed, the file is touched below when it is purged.

Reply:
Yes but what about ten days after we've purged it? After that it will be checked every time which is expensive. Should we do some sort of archiving after a long time (ex 30 days after purging).

Reply:
The problem is that we can't query the db efficiently. For now, this is not a big issue. It happens once a day, and it will finish in minutes, even with thousands of simulations. Let's get this out the door, and worry about the efficiency problem later. 

We will eventually want to denormalize some state in SQL so we can write a query against it. However, this will probably mean replacing sqlite with postgres for production, which I think we can wait to do for some time.